fedimint-server/src/consensus/mod.rs

```diff
@@ -286,67 +280,65 @@ impl FedimintConsensus {
                 // in_scope to make sure that no await is in the middle of the span
                 let _enter = span.in_scope(|| {
                     trace!(?transaction);
-                    batch_tx.append_maybe_delete(ProposedTransactionKey(transaction.tx_hash()));
+                    dbtx.maybe_remove_entry(&ProposedTransactionKey(transaction.tx_hash()))
+                        .expect("DB Error");
 
                     // TODO: use borrowed transaction
-                    match self.process_transaction(
-                        batch_tx.subtransaction(),
-                        transaction.clone(),
-                        &caches,
-                    ) {
+                    match self.process_transaction(&mut dbtx, transaction.clone(), &caches) {
                         Ok(()) => {
-                            batch_tx.append_insert(
-                                AcceptedTransactionKey(transaction.tx_hash()),
-                                AcceptedTransaction { epoch, transaction },
-                            );
+                            dbtx.insert_entry(
+                                &AcceptedTransactionKey(transaction.tx_hash()),
+                                &AcceptedTransaction { epoch, transaction },
+                            )
+                            .expect("DB Error");
                         }
                         Err(error) => {
+                            dbtx.rollback();
```

Comment:
Should not calling `commit_tx` and droping the `TransactionDatabase` just imply rollback?

Reply:
Perhaps this is a bad name. Rollback doesn't rollback the transaction, it simply resets the transactions state back to the savepoint. You are correct that simply not commiting it implies rollback, but that's not what this function does. I'll update the name.

Reply:
Renamed to rollback_tx_to_savepoint